code/modules/food_and_drinks/kitchen_machinery/microwave.dm

```diff
@@ -230,32 +230,22 @@
 		muck_finish()
 		return
 
-	else if (has_extra_item())
-		if (!microwaving(4))
+	else
+		if(has_extra_item() && prob(min(dirty*5,100)) && !microwaving(4))
 			broke()
 			return
-
-		broke()
-		return
-
-	else
-
+				
 		if(!microwaving(10))
 			abort()
 			return
 		stop()
 
-		for(var/obj/item/weapon/reagent_containers/food/snacks/F in contents)
-			if(F.cooked_type)
-				var/obj/item/weapon/reagent_containers/food/snacks/S = new F.cooked_type (get_turf(src))
-				F.initialize_cooked_food(S, efficiency)
-				feedback_add_details("food_made","[F.type]")
-			else
-				new /obj/item/weapon/reagent_containers/food/snacks/badrecipe(src)
+		for(var/obj/O in contents)
+			O.microwave_act(src)
+			if(!istype(O,/obj/item/weapon/reagent_containers/food) && !istype(O, /obj/item/weapon/grown))
```

Comment:
You can remove these three lines if you make the /atom/proc/microwave_act() do this by default.

Then in every override, you can use ..() to make the microwave dirty